android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt

```diff
@@ -62,8 +76,19 @@ class QuestionnaireActivity : BaseMultiLanguageActivity() {
   private var questionnaire: Questionnaire? = null
   private var alertDialog: AlertDialog? = null
 
+  private lateinit var fusedLocationClient: FusedLocationProviderClient
+  private val loc = org.hl7.fhir.r4.model.Location()
+
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
+
+    val applicationConfiguration = viewModel.applicationConfiguration
+
+    if (applicationConfiguration.logQuestionnaireLocation) {
+      fusedLocationClient = LocationServices.getFusedLocationProviderClient(this)
+      getLocation()
```

Comment:
is this running in the foreground or background? If there is not a GPS fix, what happens?

Reply:
@pld There are some updates to the flow.

The GPS location request is made in the background when a questionnaire is launched. This is done after config, permissions and location services checks are complete. On submit, the GPS location - if available - is used to create the location resource.